### PR TITLE
Fixing some issues with proxies

### LINKED ISSUE #3072: Crash using new proxy syntax
Input:

```haskell
module Main where
class C a b | a -> b
instance c :: C Int String
go :: forall a b. C a b => @a -> b -> b
go _ b = b
foo :: _
foo = go @Int "hello"
```

Output:

```
purs: An internal error occurred during compilation: "ctorModules: unqualified type name"
Please report this at https://github.com/purescript/purescript/issues
CallStack (from HasCallStack):
  error, called at src/Language/PureScript/Crash.hs:24:3 in purescript-0.11.6-4vkhJJqnGWo13TgqQuOCFm:Language.PureScript.Crash
  internalError, called at src/Language/PureScript/TypeChecker/Entailment.hs:201:59 in purescript-0.11.6-4vkhJJqnGWo13TgqQuOCFm:Language.PureScript.TypeChecker.Entailment
purs: thread blocked indefinitely in an MVar operation
* ERROR: Subcommand terminated with exit code 1
```

### PR DESCRIPTION
**Pull request by LiamGoodacre**
Fixes #3072

### CODE REVIEW

File: src/Language/PureScript/TypeChecker/Subsumption.hs [hunk 1]
```diff
@@ -94,6 +94,8 @@ subsumes' mode (KindedType ty1 _) ty2 =
   subsumes' mode ty1 ty2
 subsumes' mode ty1 (KindedType ty2 _) =
   subsumes' mode ty1 ty2
+subsumes' mode (ProxyType ty1) (ProxyType ty2) =
```
Comment (paf31) [inline src/Language/PureScript/TypeChecker/Subsumption.hs]: We shouldn't have proxies for polymorphic things, so I don't know why we need this. In any case, I don't think it's true that `Proxy` is necessarily covariant.

File: src/Language/PureScript/TypeChecker/Types.hs [hunk 2]
```diff
@@ -732,6 +733,11 @@ check' val kt@(KindedType ty kind) = do
 check' (PositionedValue pos c val) ty = warnAndRethrowWithPositionTC pos $ do
   TypedValue t v ty' <- check' val ty
   return $ TypedValue t (PositionedValue pos c v) ty'
+check' p@(Proxy ty1) ty2 = do
```
Comment (paf31) [inline src/Language/PureScript/TypeChecker/Types.hs]: I don't think we need this case. Falling through to `infer` should work.

**Comment by paf31** [comment]
Looks good! Do we skolemize the type under the `Proxy` constructor yet in `skolemizeTypesInValue`? We probably should.

**Comment by LiamGoodacre** [comment]
@paf31, thanks for reviewing so quickly.  I was sure there'd be some redundant changes - I changed lots of stuff until it worked and hadn't worked out which bits actually mattered.

**Pull request merged**